packages/eslint-plugin/src/rules/no-base-to-string.ts

```diff
@@ -94,7 +94,7 @@ export default createRule<Options, MessageIds>({
       node: TSESTree.Node,
       type: ts.Type,
     ): void {
-      const certainty = collectJoinCertainty(type);
+      const certainty = collectArrayToStringCertainty(type);
```

Comment:
It made sense to me to rename the function since it's not being used specifically for `array.join()` checks.